### ISSUE TITLE
initialize_async_ioctl_operation leaks 56 bytes of memory when fault injection is enabled

### ISSUE THREAD

**Issue by saxena-anurag** [microsoft/ebpf-for-windows#2251]
Memory leaked:

```
    [0]              : "Leak of 56 bytes at 2553030341184." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [1]              : "    ebpf_allocate + 190 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\platform\user\ebpf_platform_user.cpp:398)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [2]              : "    initialize_async_ioctl_operation + 56 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api_common\device_helper.cpp:171)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [3]              : "    ebpf_ring_buffer_map_subscribe + 2518 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api\ebpf_api.cpp:3760)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [4]              : "    ring_buffer__new + 161 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api\libbpf_map.cpp:319)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [5]              : "    ring_buffer_api_test_helper + 266 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\libs\common\common_tests.cpp:188)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [6]              : "    bindmonitor_ring_buffer_test + 2505 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\end_to_end.cpp:910)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [7]              : "    CATCH2_INTERNAL_TEST_27 + 16 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\end_to_end.cpp:983)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [8]              : "    Catch::`anonymous namespace'::TestInvokerAsFunction::invoke + 18 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_test_registry.cpp:58)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [9]              : "    Catch::TestCaseHandle::invoke + 33 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_test_case_info.hpp:116)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [10]             : "    Catch::RunContext::invokeActiveTestCase + 71 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp:544)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [11]             : "    Catch::RunContext::runCurrentTest + 592 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp:509)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [12]             : "    Catch::RunContext::runTest + 738 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp:239)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [13]             : "    Catch::`anonymous namespace'::TestGroup::execute + 243 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp:111)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [14]             : "    Catch::Session::runInternal + 1008 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp:336)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [15]             : "    Catch::Session::run + 80 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp:264)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [16]             : "    Catch::Session::run<char> + 82 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.hpp:41)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [17]             : "    main + 97 (D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_main.cpp:36)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [18]             : "    invoke_main + 57 (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:79)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [19]             : "    __scrt_common_main_seh + 302 (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [20]             : "    __scrt_common_main + 14 (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:331)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [21]             : "    mainCRTStartup + 14 (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp:17)." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [22]             : "    BaseThreadInitThunk + 16." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [23]             : "    RtlUserThreadStart + 43." [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [24]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [25]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [26]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [27]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [28]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [29]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [30]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [31]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [32]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [33]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [34]             : "" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
```

Fault injection stack:

```
    [0]              : "ebpf_allocate + 72 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\platform\user\ebpf_platform_user.cpp 384" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [1]              : "ebpf_allocate_with_tag + 24 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\platform\user\ebpf_platform_user.cpp 407" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [2]              : "ebpf_ring_buffer_map_async_query + 334 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\execution_context\ebpf_maps.c 1948" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [3]              : "_ebpf_core_protocol_ring_buffer_map_async_query + 228 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\execution_context\ebpf_core.c 1781" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [4]              : "ebpf_core_invoke_protocol_handler + 848 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\execution_context\ebpf_core.c 2426" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [5]              : "GlueDeviceIoControl + 693 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\test_helper.cpp 448" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [6]              : "std::invoke<bool (__cdecl*&)(void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *),void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *> + 161 C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\include\type_traits 1587" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [7]              : "std::_Invoker_ret<int>::_Call<bool (__cdecl*&)(void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *),void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *> + 115 C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\include\functional 672" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [8]              : "std::_Func_impl_no_alloc<bool (__cdecl*)(void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *),int,void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *>::_Do_call + 269 C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\include\functional 834" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [9]              : "std::_Func_class<int,void *,unsigned long,void *,unsigned long,void *,unsigned long,unsigned long *,_OVERLAPPED *>::operator() + 353 C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\include\functional 875" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [10]             : "Platform::DeviceIoControl + 115 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\thunk\mock\mock.cpp 32" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [11]             : "invoke_ioctl<_ebpf_operation_ring_buffer_map_async_query_request,_ebpf_operation_ring_buffer_map_async_query_reply> + 193 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api_common\device_helper.hpp 99" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [12]             : "_ebpf_ring_buffer_map_async_query_completion + 2144 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api\ebpf_api.cpp 3683" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [13]             : "`initialize_async_ioctl_operation'::`2'::<lambda_1>::operator() + 79 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api_common\device_helper.cpp 193" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [14]             : "`initialize_async_ioctl_operation'::`2'::<lambda_1>::<lambda_invoker_cdecl> + 58 D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\api_common\device_helper.cpp 193" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
    [15]             : "TpAllocTimer + 978" [Type: std::basic_string<char,std::char_traits<char>,std::allocator<char> >]
```

**Comment by shpalani** [comment]
Crashdump

0:000> k
 # Child-SP          RetAddr               Call Site
00 0000000b`f9b409f8 00007ffc`ae91ffdc     ntdll!ZwWaitForMultipleObjects+0x14 [minkernel\ntdll\daytona\objfre\amd64\usrstubs.asm @ 947] 
01 0000000b`f9b40a00 00007ffc`ae91fede     KERNELBASE!WaitForMultipleObjectsEx+0xec [minkernel\kernelbase\synch.c @ 1551] 
02 0000000b`f9b40cf0 00007ffc`af8d2301     KERNELBASE!WaitForMultipleObjects+0xe [minkernel\kernelbase\synch.c @ 1403] 
03 0000000b`f9b40d30 00007ffc`af8d1ecd     kernel32!WerpReportFaultInternal+0x425
04 0000000b`f9b40e50 00007ffc`af8d1e3e     kernel32!WerpReportFault+0x81
05 0000000b`f9b40e90 00007ffc`ae967db4     kernel32!BasepReportFault+0x1e
06 0000000b`f9b40ec0 00007ffc`b1168585     KERNELBASE!UnhandledExceptionFilter+0x344 [minkernel\kernelbase\xcpt.c @ 685] 
07 (Inline Function) --------`--------     ntdll!RtlpThreadExceptionFilter+0x9b [minkernel\ntdll\rtlstrt.c @ 956] 
08 0000000b`f9b40fe0 00007ffc`b114e726     ntdll!RtlUserThreadStart$filt$0+0xac [minkernel\ntdll\rtlstrt.c @ 1153] 
09 0000000b`f9b41020 00007ffc`b116438f     ntdll!__C_specific_handler+0x96 [minkernel\crts\crtw32\misc\riscchandler.c @ 446] 
0a 0000000b`f9b41090 00007ffc`b10f170e     ntdll!RtlpExecuteHandlerForException+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 132] 
0b 0000000b`f9b410c0 00007ffc`b110f2c3     ntdll!RtlDispatchException+0x26e [minkernel\ntos\rtl\amd64\exdsptch.c @ 653] 
0c 0000000b`f9b41800 00007ffc`ae8fef5c     ntdll!RtlRaiseException+0x163 [minkernel\ntos\rtl\amd64\raise.c @ 215] 
0d 0000000b`f9b42620 00007ff7`b4bf3d35     KERNELBASE!RaiseException+0x6c [minkernel\kernelbase\xcpt.c @ 936] 
0e 0000000b`f9b42700 00007ffc`91046e31     unit_tests!_wer_report_test::_terminate_hook+0x25 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\libs\util\wer_report.hpp @ 91] 
0f 0000000b`f9b42730 00007ffc`91045610     ucrtbased!_VCrtDbgReportW+0x841 [minkernel\crts\ucrt\src\appcrt\misc\dbgrptt.cpp @ 613] 
10 0000000b`f9b4a870 00007ffc`91032fa1     ucrtbased!_CrtDbgReportW+0x60 [minkernel\crts\ucrt\src\appcrt\misc\dbgrpt.cpp @ 278] 
11 0000000b`f9b4a8d0 00007ffc`91033153     ucrtbased!issue_debug_notification+0x31 [minkernel\crts\ucrt\src\appcrt\internal\report_runtime_error.cpp @ 25] 
12 0000000b`f9b4a920 00007ffc`9104ae2d     ucrtbased!__acrt_report_runtime_error+0x13 [minkernel\crts\ucrt\src\appcrt\internal\report_runtime_error.cpp @ 154] 
13 0000000b`f9b4a980 00007ffc`91050c6f     ucrtbased!abort+0x1d [minkernel\crts\ucrt\src\appcrt\startup\abort.cpp @ 61] 
14 0000000b`f9b4a9c0 00007ffc`9104eba1     ucrtbased!common_assert_to_stderr<wchar_t>+0xbf [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 187] 
15 0000000b`f9b4aa20 00007ffc`910518af     ucrtbased!common_assert<wchar_t>+0x41 [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 415] 
16 0000000b`f9b4aa60 00007ff7`b4d7b790     ucrtbased!_wassert+0x2f [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 444] 
17 0000000b`f9b4aa90 00007ff7`b4d57f69     unit_tests!_ebpf_leak_detector::dump_leaks+0x7c0 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\platform\user\ebpf_leak_detector.cpp @ 69] 
18 0000000b`f9b4aff0 00007ff7`b4cffdb6     unit_tests!ebpf_platform_terminate+0x59 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\platform\user\ebpf_platform_user.cpp @ 359] 
19 0000000b`f9b4b030 00007ff7`b4b7c232     unit_tests!ebpf_core_terminate+0x66 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\libs\execution_context\ebpf_core.c @ 268] 
1a 0000000b`f9b4b060 00007ff7`b4f5c87c     unit_tests!_test_helper_end_to_end::~_test_helper_end_to_end+0x412 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\test_helper.cpp @ 692] 
1b 0000000b`f9b4b390 00007ffc`95e81030     unit_tests!`bindmonitor_ring_buffer_test'::`1'::dtor$0+0x1c
1c 0000000b`f9b4b3d0 00007ffc`95e842a7     VCRUNTIME140_1D!_CallSettingFrame+0x20 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\amd64\handlers.asm @ 50] 
1d 0000000b`f9b4b400 00007ffc`95e8668b     VCRUNTIME140_1D!__FrameHandler4::FrameUnwindToState+0x2b7 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 1146] 
1e 0000000b`f9b4b510 00007ffc`95e82cd2     VCRUNTIME140_1D!__FrameHandler4::FrameUnwindToEmptyState+0x4b [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 234] 
1f 0000000b`f9b4b550 00007ffc`95e82f57     VCRUNTIME140_1D!__InternalCxxFrameHandler<__FrameHandler4>+0x282 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 354] 
20 0000000b`f9b4b620 00007ffc`95e86ddb     VCRUNTIME140_1D!__InternalCxxFrameHandlerWrapper<__FrameHandler4>+0x67 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 234] 
21 0000000b`f9b4b680 00007ff7`b4db2b40     VCRUNTIME140_1D!__CxxFrameHandler4+0xfb [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 304] 
22 0000000b`f9b4b710 00007ffc`b116440f     unit_tests!__GSHandlerCheck_EH4+0x90 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\gs\amd64\gshandlereh4.cpp @ 73] 
23 0000000b`f9b4b760 00007ffc`b10eef7c     ntdll!RtlpExecuteHandlerForUnwind+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 254] 
24 0000000b`f9b4b790 00007ffc`95e86a5f     ntdll!RtlUnwindEx+0x2fc [minkernel\ntos\rtl\amd64\exdsptch.c @ 1521] 
25 0000000b`f9b4bec0 00007ffc`95e81c1e     VCRUNTIME140_1D!__FrameHandler4::UnwindNestedFrames+0x1af [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 685] 
26 0000000b`f9b4bfc0 00007ffc`95e8218b     VCRUNTIME140_1D!CatchIt<__FrameHandler4>+0x12e [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 1370] 
27 0000000b`f9b4c060 00007ffc`95e82ec5     VCRUNTIME140_1D!FindHandler<__FrameHandler4>+0x55b [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 679] 
28 0000000b`f9b4c240 00007ffc`95e82f57     VCRUNTIME140_1D!__InternalCxxFrameHandler<__FrameHandler4>+0x475 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 397] 
29 0000000b`f9b4c310 00007ffc`95e86ddb     VCRUNTIME140_1D!__InternalCxxFrameHandlerWrapper<__FrameHandler4>+0x67 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 234] 
2a 0000000b`f9b4c370 00007ff7`b4db2b40     VCRUNTIME140_1D!__CxxFrameHandler4+0xfb [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 304] 
2b 0000000b`f9b4c400 00007ffc`b116438f     unit_tests!__GSHandlerCheck_EH4+0x90 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\gs\amd64\gshandlereh4.cpp @ 73] 
2c 0000000b`f9b4c450 00007ffc`b10f170e     ntdll!RtlpExecuteHandlerForException+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 132] 
2d 0000000b`f9b4c480 00007ffc`b110f2c3     ntdll!RtlDispatchException+0x26e [minkernel\ntos\rtl\amd64\exdsptch.c @ 653] 
2e 0000000b`f9b4cbc0 00007ffc`ae8fef5c     ntdll!RtlRaiseException+0x163 [minkernel\ntos\rtl\amd64\raise.c @ 215] 
2f 0000000b`f9b4d9e0 00007ffc`95e9b760     KERNELBASE!RaiseException+0x6c [minkernel\kernelbase\xcpt.c @ 936] 
30 0000000b`f9b4dac0 00007ff7`b4bffea8     VCRUNTIME140D!_CxxThrowException+0x120 [D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\throw.cpp @ 75] 
31 0000000b`f9b4db50 00007ff7`b4bfc25d     unit_tests!Catch::throw_test_failure_exception+0x28 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_test_failure_exception.cpp @ 21] 
32 0000000b`f9b4db90 00007ff7`b4d993d2     unit_tests!Catch::AssertionHandler::complete+0x4d [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_assertion_handler.cpp @ 52] 
33 0000000b`f9b4dbd0 00007ff7`b4b26e69     unit_tests!ring_buffer_api_test_helper+0x532 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\libs\common\common_tests.cpp @ 198] 
34 0000000b`f9b4df70 00007ff7`b4b098e0     unit_tests!bindmonitor_ring_buffer_test+0x9c9 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\end_to_end.cpp @ 910] 
35 0000000b`f9b4e850 00007ff7`b4bfd282     unit_tests!CATCH2_INTERNAL_TEST_25+0x10 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\tests\end_to_end\end_to_end.cpp @ 983] 
36 0000000b`f9b4e880 00007ff7`b4c0c2a1     unit_tests!Catch::`anonymous namespace'::TestInvokerAsFunction::invoke+0x12 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_test_registry.cpp @ 58] 
37 0000000b`f9b4e8b0 00007ff7`b4c0c2f7     unit_tests!Catch::TestCaseHandle::invoke+0x21 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_test_case_info.hpp @ 116] 
38 0000000b`f9b4e8e0 00007ff7`b4c0cee0     unit_tests!Catch::RunContext::invokeActiveTestCase+0x47 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 544] 
39 0000000b`f9b4e930 00007ff7`b4c0d352     unit_tests!Catch::RunContext::runCurrentTest+0x250 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 509] 
3a 0000000b`f9b4ecb0 00007ff7`b4f1a283     unit_tests!Catch::RunContext::runTest+0x2e2 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 239] 
3b 0000000b`f9b4f190 00007ff7`b4f1b220     unit_tests!Catch::`anonymous namespace'::TestGroup::execute+0xf3 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp @ 111] 
3c 0000000b`f9b4f310 00007ff7`b4f1adc0     unit_tests!Catch::Session::runInternal+0x3f0 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp @ 336] 
3d 0000000b`f9b4f790 00007ff7`b4db6222     unit_tests!Catch::Session::run+0x50 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.cpp @ 264] 
3e 0000000b`f9b4f7d0 00007ff7`b4db6291     unit_tests!Catch::Session::run<char>+0x52 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\catch_session.hpp @ 41] 
3f 0000000b`f9b4f810 00007ff7`b4db3179     unit_tests!main+0x61 [D:\a\ebpf-for-windows-1\ebpf-for-windows-1\external\Catch2\src\catch2\internal\catch_main.cpp @ 36] 
40 0000000b`f9b4fa30 00007ff7`b4db30ce     unit_tests!invoke_main+0x39 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 79] 
41 0000000b`f9b4fa80 00007ff7`b4db2f8e     unit_tests!__scrt_common_main_seh+0x12e [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 288] 
42 0000000b`f9b4faf0 00007ff7`b4db31ee     unit_tests!__scrt_common_main+0xe [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 331] 
43 0000000b`f9b4fb20 00007ffc`af8c4de0     unit_tests!mainCRTStartup+0xe [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp @ 17] 
44 0000000b`f9b4fb50 00007ffc`b113e3db     kernel32!BaseThreadInitThunk+0x10
45 0000000b`f9b4fb80 00000000`00000000     ntdll!RtlUserThreadStart+0x2b [minkernel\ntdll\rtlstrt.c @ 1152]

**Comment by shpalani** [comment]
This memory leak is fixed by #2304.

**Comment by shpalani** [comment]
bindmonitor_ring_buffer_test() is called by the following test cases: 
bindmonitor-ringbuf-native
bindmonitor-ringbuf-jit
bindmonitor-ringbuf-interpret

Deallocation:
ring_buffer__new  --> ring_buffer__free
_ring_buffer_test_event_context --> unsubscribe() --> ring_buffer__free
